python/utils/bootstrap_utilities.py

```diff
@@ -391,3 +392,128 @@ def handle_error(data, proxy=None):
+    or when direct access to the tk-desktop logging system is not available.
+
+    :param data: A dictionary containing proxy connection details, including the path
+                 to the RPC library and proxy authentication information.
+    :param message: The log message to send to tk-desktop.
+    """
+    if data["core_python_path"] not in sys.path:
+        sys.path.insert(0, data["core_python_path"])
+    import sgtk
+
+    proxy = _create_proxy(data)
+    handler = ProxyLoggingHandler(proxy)
+    sgtk.LogManager().initialize_custom_handler(handler)
+    logger = sgtk.LogManager.get_logger(__file__)
+    logger.error(message)
+    proxy.close()
+
+
+def execute_pre_initialization_hook(data):
+    """
+    Executes the 'hook_pre_initialization' hook.
+
+    This function dynamically loads and executes the 'hook_pre_initialization' hook,
+    which is responsible for performing early initialization tasks before the engine
+    is fully initialized. These tasks may include importing necessary libraries,
+    setting environment variables, or preparing the environment to ensure compatibility
+    with PySide6 and other dependencies.
+
+    If a custom hook is not defined in the advanced configuration (`tk-desktop.yml`),
+    the function falls back to dynamically loading the default hook located in the
+    'hooks' directory of the project.
+
+    The function follows these steps:
+    1. Reads the `info.yml` file to retrieve the default hook name for pre-initialization.
+    2. Reads the `tk-desktop.yml` file from the advanced configuration to check if a
+       custom hook is defined for the project.
+    3. Resolves the path to the custom hook if defined, or falls back to the default hook.
+    4. Dynamically loads the hook module and executes its `PreInitialization.execute` method.
+
+    :param data: A dictionary containing configuration and runtime details, including:
+                 - `config_path`: The root path of the advanced configuration.
+                 - `proxy_data`: Proxy connection details for logging.
+                 - `rpc_lib_path`: Path to the RPC library for communication.
+    """
+
+    try:
+        if data["core_python_path"] not in sys.path:
+            sys.path.insert(0, data["core_python_path"])
+
+        from tank_vendor import yaml
+
+        # Step 1: Read info.yml to get the default hook name
+        info_yml_path = os.path.join(os.path.dirname(__file__), "..", "..", "info.yml")
+        info = {}
+        with open(info_yml_path, "rt") as fh:
+            info = yaml.load(fh, Loader=yaml.FullLoader)
+        default_hook_name = (
+            info.get("configuration", {})
+            .get("hook_pre_initialization", {})
+            .get("default_value", "")
+        )
+
+        # Step 2: Read tk-desktop.yml to check for a custom hook
+        tk_desktop_yml_path = os.path.join(
+            data.get("config_path"),
```

Comment:
@eduardoChaucaGallegos using this `config_path`, you can easily implement the missing `{config}` pattern in this method, isn't it?